spec/controllers/webhook_controller_spec.rb

```diff
@@ -51,4 +51,37 @@ def webhook(params)
 
     expect { webhook id: 'evt_charge_succeeded' }.to raise_error(Stripe::StripeError, /testing/)
   end
+  
+  context "with an authentication secret" do
+    def webhook(secret, params)
+      if secret
+        request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Basic.encode_credentials('user', secret)
```

Comment:
Overriding `webhook` in this context reduces duplication but makes the tests a bit harder to reason about. I'd rather repeat `request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Basic.encode_credentials('user', secret)` is each test for clarity.